src/redux/sagas/motions/finalizeMotion.ts

```diff
@@ -1,54 +1,47 @@
 import { call, fork, put, takeEvery } from 'redux-saga/effects';
-import { AnyVotingReputationClient, ClientType } from '@colony/colony-js';
-import { AddressZero } from '@ethersproject/constants';
-import { BigNumber } from 'ethers';
+import { ClientType } from '@colony/colony-js';
 
 import { ActionTypes } from '../../actionTypes';
 import { AllActions, Action } from '../../types/actions';
-import {
-  putError,
-  takeFrom,
-  updateMotionValues,
-  getColonyManager,
-} from '../utils';
+import { putError, takeFrom } from '../utils';
 
 import {
   createTransaction,
   createTransactionChannels,
   getTxChannel,
 } from '../transactions';
-import { transactionReady, transactionUpdateGas } from '../../actionCreators';
+import { transactionReady } from '../../actionCreators';
 
 function* finalizeMotion({
   meta,
-  payload: { userAddress, colonyAddress, motionId },
+  payload: { /* userAddress */ colonyAddress, motionId },
 }: Action<ActionTypes.MOTION_FINALIZE>) {
   const txChannel = yield call(getTxChannel, meta.id);
   try {
-    const colonyManager = yield getColonyManager();
-    const { provider } = colonyManager;
-    const colonyClient = yield colonyManager.getClient(
-      ClientType.ColonyClient,
-      colonyAddress,
-    );
-    const votingReputationClient: AnyVotingReputationClient =
-      yield colonyManager.getClient(
-        ClientType.VotingReputationClient,
-        colonyAddress,
-      );
-    const motion = yield votingReputationClient.getMotion(motionId);
-
-    const networkEstimate = yield provider.estimateGas({
-      from: votingReputationClient.address,
-      to:
-        /*
-         * If the motion target is 0x000... then we pass in the colony's address
-         */
-        motion.altTarget === AddressZero
-          ? colonyClient.address
-          : motion.altTarget,
-      data: motion.action,
-    });
+    // const colonyManager = yield getColonyManager();
+    // const { provider } = colonyManager;
+    // const colonyClient = yield colonyManager.getClient(
+    //   ClientType.ColonyClient,
+    //   colonyAddress,
+    // );
+    // const votingReputationClient: AnyVotingReputationClient =
+    //   yield colonyManager.getClient(
+    //     ClientType.VotingReputationClient,
+    //     colonyAddress,
+    //   );
+    // const motion = yield votingReputationClient.getMotion(motionId);
+
+    // const networkEstimate = yield provider.estimateGas({
```

Comment:
No please don't :)

Every finalize transaction is (more or less) unique, meaning it will have different gas requirements depending on the underlying action the motion will execute once finalized.

This estimate call allows us to get a sense of the gas required for this transaction. Atop of it we add a bit more ensure that we account for any mismatches in gas between the time we estimate and the time we actually execute (kine 53-55)

This was actually a big problem when originally implementing finalize motions, and it took a while for us to figure out a way to make it work for all "current" motions

So please uncomment all these lines